Setting `CMAKE_CXX_COMPILER_LAUNCHER` breaks `kokkos_launch_compiler`

Attempting to compile a CMake project depending on Kokkos with the CMake variable `CMAKE_CXX_COMPILER_LAUNCHER` set to `ccache` leads to build failures (at least with CUDA enabled). Build output with `KOKKOS_DEBUG_LAUNCH_COMPILER=1`:
```
> ninja -j1 tests                                                                                                                              
                                                                                                                                               
[1/17] Building CXX object tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o                                                               
FAILED: tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o                                                                                  
/apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/kokkos-develop-6uzxrzjk2cd7e3t5nmbgfwhek5xsluty/bin/kokkos_launch_compile
r /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/kokkos-develop-6uzxrzjk2cd7e3t5nmbgfwhek5xsluty/bin/nvcc_wrapper /apps/
daint/UES/simbergm/spack/lib/spack/env/gcc/g++ ccache /apps/daint/UES/simbergm/spack/lib/spack/env/gcc/g++ -DHPX_APPLICATION_EXPORTS -DHPX_APPL
ICATION_NAME_DEFAULT=asynchrony_test -DHPX_KOKKOS_CUDA_FUTURE_TYPE=0 -DHPX_PREFIX_DEFAULT=\"/apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-
haswell/gcc-11.2.0/hpx-1.7.1-2iro5e2o25fyrpku446o6tkkjulw4bba\" -DKOKKOS_DEPENDENCE -I/apps/daint/UES/simbergm/src/hpx-kokkos/src -isystem /app
s/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/hpx-1.7.1-2iro5e2o25fyrpku446o6tkkjulw4bba/include -isystem /apps/daint/UES/s
imbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/cuda-11.6.0-2ktngoaeb3mpv22nf2hlyzpirx6enu4v/include -isystem /apps/daint/UES/simbergm/spa
ck/opt/spack/cray-cnl7-haswell/gcc-11.2.0/boost-1.78.0-tysrpumihotokpynwc6mf2w2bz4xe3on/include -isystem /apps/daint/UES/simbergm/spack/opt/spa
ck/cray-cnl7-haswell/gcc-11.2.0/asio-1.20.0-v6ddoyrwskgpwyfrsf3kmrl3ycfnlr5f/include -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl
7-haswell/gcc-11.2.0/hwloc-2.7.0-44knj6u72gqwytzghcp33jrmjqeveqsl/include -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/g
cc-11.2.0/kokkos-develop-6uzxrzjk2cd7e3t5nmbgfwhek5xsluty/include -g -pthread -march=core-avx2 -mtune=core-avx2 -expt-extended-lambda -Wext-lam
bda-captures-this -expt-relaxed-constexpr -arch=sm_60 -std=c++14 -MD -MT tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o -MF tests/CMakeF
iles/asynchrony_test.dir/asynchrony.cpp.o.d -o tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o -c /apps/daint/UES/simbergm/src/hpx-kokkos
/tests/asynchrony.cpp                                                                                                                          
##### kokkos_launch_compiler executing: "ccache /apps/daint/UES/simbergm/spack/lib/spack/env/gcc/g++ -DHPX_APPLICATION_EXPORTS -DHPX_APPLICATIO
N_NAME_DEFAULT=asynchrony_test -DHPX_KOKKOS_CUDA_FUTURE_TYPE=0 -DHPX_PREFIX_DEFAULT="/apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell
/gcc-11.2.0/hpx-1.7.1-2iro5e2o25fyrpku446o6tkkjulw4bba" -DKOKKOS_DEPENDENCE -I/apps/daint/UES/simbergm/src/hpx-kokkos/src -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/hpx-1.7.1-2iro5e2o25fyrpku446o6tkkjulw4bba/include -isystem /apps/daint/UES/simbergm/
spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/cuda-11.6.0-2ktngoaeb3mpv22nf2hlyzpirx6enu4v/include -isystem /apps/daint/UES/simbergm/spack/opt/s
pack/cray-cnl7-haswell/gcc-11.2.0/boost-1.78.0-tysrpumihotokpynwc6mf2w2bz4xe3on/include -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-
cnl7-haswell/gcc-11.2.0/asio-1.20.0-v6ddoyrwskgpwyfrsf3kmrl3ycfnlr5f/include -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.0/hwloc-2.7.0-44knj6u72gqwytzghcp33jrmjqeveqsl/include -isystem /apps/daint/UES/simbergm/spack/opt/spack/cray-cnl7-haswell/gcc-11.2.
0/kokkos-develop-6uzxrzjk2cd7e3t5nmbgfwhek5xsluty/include -g -pthread -march=core-avx2 -mtune=core-avx2 -expt-extended-lambda -Wext-lambda-capt
ures-this -expt-relaxed-constexpr -arch=sm_60 -std=c++14 -MD -MT tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o -MF tests/CMakeFiles/asy
nchrony_test.dir/asynchrony.cpp.o.d -o tests/CMakeFiles/asynchrony_test.dir/asynchrony.cpp.o -c /apps/daint/UES/simbergm/src/hpx-kokkos/tests/a
synchrony.cpp"... #####                                                                                                                        
g++: error: unrecognized command-line option '-Wext-lambda-captures-this'                                                                      g++: error: unrecognized command-line option '-arch=sm_60'                                                                                     
ninja: build stopped: subcommand failed.
```
Abbreviated build command: `kokkos_launch_compiler nvcc_wrapper g++ ccache g++ ...`. I suppose the extra `ccache` in between `g++` and `g++` throws off the detection for when to forward all arguments and when not to.

Is this a bug or "unsupported"? I found https://github.com/lammps/lammps/pull/1565 which seems to indicate that we can possibly do something to make this work outside of Kokkos, but I wanted to first check if this might be something that can be fixed here.

--- a/cmake/kokkos_compiler_id.cmake
+++ b/cmake/kokkos_compiler_id.cmake
@@ -37,9 +37,9 @@ IF(Kokkos_ENABLE_CUDA)
         PATHS           ${PROJECT_SOURCE_DIR}
         PATH_SUFFIXES   bin)
 
-    # check if compiler was set to nvcc_wrapper
+    # Check if compiler was set to nvcc_wrapper
     kokkos_internal_have_compiler_nvcc(${CMAKE_CXX_COMPILER})
-    # if launcher was found and nvcc_wrapper was not specified as
+    # If launcher was found and nvcc_wrapper was not specified as
     # compiler and `CMAKE_CXX_COMPILIER_LAUNCHER` is not set, set to use launcher.
     # Will ensure CMAKE_CXX_COMPILER is replaced by nvcc_wrapper
     IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)

--- a/cmake/kokkos_compiler_id.cmake
+++ b/cmake/kokkos_compiler_id.cmake
@@ -40,12 +40,13 @@ IF(Kokkos_ENABLE_CUDA)
     # check if compiler was set to nvcc_wrapper
     kokkos_internal_have_compiler_nvcc(${CMAKE_CXX_COMPILER})
     # if launcher was found and nvcc_wrapper was not specified as
-    # compiler AND `CMAKE_CXX_COMPILIER_LAUNCHER` is not set, set to use launcher. Will ensure CMAKE_CXX_COMPILER
-    # is replaced by nvcc_wrapper
-    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT
-       CMAKE_CXX_COMPILER_LAUNCHER AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
-			MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or
+    # compiler and `CMAKE_CXX_COMPILIER_LAUNCHER` is not set, set to use launcher.
+    # Will ensure CMAKE_CXX_COMPILER is replaced by nvcc_wrapper
+    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+      IF(CMAKE_CXX_COMPILER_LAUNCHER)
+      MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or
 clang++!")
+      ENDIF()
       # the first argument to launcher is always the C++ compiler defined by cmake
       # if the second argument matches the C++ compiler, it forwards the rest of the
       # args to nvcc_wrapper

--- a/cmake/kokkos_compiler_id.cmake
+++ b/cmake/kokkos_compiler_id.cmake
@@ -42,7 +42,8 @@ IF(Kokkos_ENABLE_CUDA)
     # if launcher was found and nvcc_wrapper was not specified as
     # compiler, set to use launcher. Will ensure CMAKE_CXX_COMPILER
     # is replaced by nvcc_wrapper
-    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+    IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT
+       CMAKE_CXX_COMPILER_LAUNCHER AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
       # the first argument to launcher is always the C++ compiler defined by cmake
       # if the second argument matches the C++ compiler, it forwards the rest of the
       # args to nvcc_wrapper

--- a/cmake/kokkos_compiler_id.cmake
+++ b/cmake/kokkos_compiler_id.cmake
@@ -40,10 +40,12 @@ IF(Kokkos_ENABLE_CUDA)
     # check if compiler was set to nvcc_wrapper
     kokkos_internal_have_compiler_nvcc(${CMAKE_CXX_COMPILER})
     # if launcher was found and nvcc_wrapper was not specified as
-    # compiler, set to use launcher. Will ensure CMAKE_CXX_COMPILER
+    # compiler AND `CMAKE_CXX_COMPILIER_LAUNCHER` is not set, set to use launcher. Will ensure CMAKE_CXX_COMPILER
     # is replaced by nvcc_wrapper
     IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT
        CMAKE_CXX_COMPILER_LAUNCHER AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
+			MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or
+clang++!")
       # the first argument to launcher is always the C++ compiler defined by cmake
       # if the second argument matches the C++ compiler, it forwards the rest of the
       # args to nvcc_wrapper

--- a/cmake/kokkos_compiler_id.cmake
+++ b/cmake/kokkos_compiler_id.cmake
@@ -44,7 +44,7 @@ IF(Kokkos_ENABLE_CUDA)
     # Will ensure CMAKE_CXX_COMPILER is replaced by nvcc_wrapper
     IF(Kokkos_COMPILE_LAUNCHER AND NOT INTERNAL_HAVE_COMPILER_NVCC AND NOT KOKKOS_CXX_COMPILER_ID STREQUAL Clang)
       IF(CMAKE_CXX_COMPILER_LAUNCHER)
-      MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or
+       MESSAGE(FATAL_ERROR "Cannot use CMAKE_CXX_COMPILER_LAUNCHER if the CMAKE_CXX_COMPILER is not able to compile CUDA code, i.e. nvcc_wrapper or
 clang++!")
       ENDIF()
       # the first argument to launcher is always the C++ compiler defined by cmake

